app/scripts/components/analysis/define/aoi-selector.tsx

```diff
@@ -50,32 +57,46 @@ interface AoiSelectorProps {
   onAoiEvent: AoiChangeListenerOverload;
 }
 
-export default function AoiSelector(props: AoiSelectorProps) {
-  const { onAoiEvent, qsFeature, aoiDrawState } = props;
+export default function AoiSelector({
+  onAoiEvent,
+  qsFeature,
+  aoiDrawState
+}: AoiSelectorProps) {
   const { selected, drawing, feature } = aoiDrawState;
   const mapRef = useRef<MapboxMapRef>(null);
 
   // Technical debt
   // Despite the query parameters support for multiple features on the aoi, the
   // AOI drawing tool only supports one.
   // Keeping just the first one.
-  const polygon: Feature<Polygon> | null = useMemo(() => {
+  const qsPolygon: Feature<Polygon> | null = useMemo(() => {
     return qsFeature
       ? { ...multiPolygonToPolygon(qsFeature), id: 'qs-feature' }
       : null;
   }, [qsFeature]);
 
-  // Use the feature from the url qs as the initial state to center the map.
+  const [currentRegionPreset, setCurrentRegionPreset] =
+    useState<RegionPreset | null>(null);
+  const onRegionPresetClick = useCallback((preset: RegionPreset) => {
+    setCurrentRegionPreset(preset);
```

Comment:
I'd say there's no need to store this in state and causing a rerender. You an trigger the aoi event and the maps reposition directly.